xgi/linalg/laplacian_matrix.py

```diff
@@ -221,16 +234,60 @@ def normalized_hypergraph_laplacian(H, sparse=True, index=False):
             "Every node must be a member of an edge to avoid divide by zero error!"
         )
 
-    D = degree_matrix(H)
-    A, rowdict = clique_motif_matrix(H, sparse=sparse, index=True)
+    if weights is not None:
+        if weights is not list:
+            raise XGIError("Edge weights must be given as a list!")
+        if len(weights) != H.num_edges:
+            raise XGIError("There must be as many edge weights as there are edges!")
+        if np.any(weights <= 0):
+            raise XGIError("Edge weights must be strictly positive!")
+
+
+    Dv = degree_matrix(H)
+    (
+        H_,
+        rowdict,
+        _            # Discard edge name-index mapping
+    ) = incidence_matrix(H, sparse=sparse, index=True)
 
     if sparse:
-        Dinvsqrt = csr_array(diags(np.power(D, -0.5)))
+        Dv_invsqrt = csr_array(diags(np.power(Dv, -0.5)))
+
+        De_inv = diags(list(
+            map(
+                lambda x: 1/x,
+                np.sum(H_, axis=0)
+            )
+        ))
```

Comment:
maybe try to simplify similarly to the non-sparse case. Also, is np.sum(H_, axis=0) still sparse?